content/blog/2017-11-10-ozunconf2017realtime.md

```diff
@@ -0,0 +1,93 @@
+---
+slug: "realtime"
+title: "2017 rOpenSci ozunconf :: Reflections and the realtime Package"
+date: 2017-11-10
+authors:
+  - name: Jonathan Carroll
+    url: https://jcarroll.com.au/
+categories: blog
+topicid: 000
+tags:
+  - community
+  - ropensci
+  - meetings
+  - unconf
+  - australia
+  - r
+  - plot
+  - realtime
+---
+
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+This year's [rOpenSci ozunconf](https://ozunconf17.ropensci.org/) was held in Melbourne, bringing together over 45 R enthusiasts from around the country and beyond. As is customary, ideas for projects were discussed in [GitHub Issues](https://github.com/ropensci/ozunconf17/issues) (41 of them by the time the unconf rolled around!) and there was no shortage of enthusiasm, interesting concepts, and varied experience.
+
+I've been to a few unconfs now and I treasure the time I get to spend with new people, new ideas, new backgrounds, new approaches, and new insights. That's not to take away from the time I get to spend with people I met at previous unconfs; I've gained great friendships and started collaborations on side projects with these wonderful people.
+
+When the call for nominations came around this year it was an easy decision. I don't have employer support to attend these things so I take time off work and pay my own way. This is my networking time, my development time, and my skill-building time. I wasn't sure what sort of project I'd be interested in but I had no doubts something would come up that sounded interesting. 
+
+As it happened, I had been playing around with a bit of code, purely out of interest and hoping to learn how [`htmlwidgets`](https://www.htmlwidgets.org/) work. The idea was to make a classic graphic equaliser visualisation like this
+
+![](/img/blog-images/2017-11-10-realtime/graphiceq.gif)
+
+using R.
+
+This presents several challenges; how can I get live audio into R, and how fast can I plot the signal? I had doubts about both parts, partly because of the way that R calls tie up the session ([for now...](https://appsilondatascience.com/blog/rstats/2017/11/01/r-promises-hands-on.html)) and partly because constructing a `ggplot2` object (without starting anything; it would be) is somewhat slow (in terms of raw audio speeds). I'd heard about `htmlwidgets` and thought there must be a way to leverage that towards my goal.
```

Comment:
the phrases in parentheses make the sentence a bit confusing "...partly because constructing a `ggplot2` object (without starting anything; it would be) is somewhat slow (in terms of raw audio speeds)..."